client/src/app/budgets/budget/budget.component.ts

```diff
@@ -14,18 +14,35 @@ import { DataChangeNotificationService } from '../../services/dataChangeNotifica
   templateUrl: './budget.component.html',
   styleUrls: ['./budget.component.css']
 })
-export class BudgetComponent {
+export class BudgetComponent implements OnInit {
   @Input() budget: Budget;
   requests: Request[];
   percentageLeft: number;
   requestStateType = RequestFilter;
   public currentYear: number;
+  budgetTypeName :  string;
 
   constructor(private requestService: RequestService,
     private modalService: NgbModal,
     public busyIndicatorService: BusyIndicatorService,
-    private dataChangeNotificationService: DataChangeNotificationService) {
-    this.currentYear = (new Date()).getFullYear();
+    private dataChangeNotificationService: DataChangeNotificationService) 
+    {
+    this.currentYear = (new Date()).getFullYear();   
+  }
+
+  ngOnInit() {
+    console.log(this.budget.type);
+    switch(this.budget.type)
+    {
+      case 0:{
+        this.budgetTypeName = 'Personal budget';
```

Comment:
Check the BudgetType class in the backend, we alreadz have API for getting the budget types with their names from the server, this whole change is unnecessary